apps/auth/src/auth.controller.ts

```diff
@@ -33,4 +38,39 @@ export class AuthController {
         const userData = await this.usersMapper.transformUserToResponseUserDto(user);
         response.send(userData);
     }
+
+    @Post('signout')
+    signOut(@Res() res: Response) {
+        this.authService.signOut(res);
+        return res.status(201).json({ msg: 'Signed out successfully!' });
+    }
+
+    @UseGuards(GoogleAuthGuard)
+    @Get('google/login')
+    handleGoogleLogin() {
+        return true;
+    }
+
+    @UseGuards(GoogleAuthGuard)
+    @Get('google/callback')
+    async handleGoogleRedirect(
+        @Req() request: Request,
+        @Res({ passthrough: true }) response: Response
+    ) {
+        const { email, firstName, lastName } = request.user as RegisterUserProfileDto;
+
+        let user: User;
+        try {
+            user = await this.usersService.getUserByEmail({ email });
+        } catch (err) {
+            const createdUser = await this.authService.registerUserProfile(
```

Comment:
Let's create a new method inside authService -> getOrRegisterUserProfile and put this try catch block there